src/expression.d

```diff
@@ -9866,6 +9866,14 @@ extern (C++) final class CallExp : UnaExp
             else if (e1.op == TOKtype && t1.isscalar())
             {
                 Expression e;
+
+                // Make sure to use the the enum type itself rather than its
+                // base type (see bugzilla 16346)
```

Comment:
We started to use direct links to bugzilla.
https://issues.dlang.org/show_bug.cgi?id=16346